sdk/agrifood/agrifood-farming-rest/review/agrifood-farming.api.md

```diff
@@ -252,77 +335,101 @@ export interface ApplicationProductDetail {
 }
 
 // @public (undocumented)
-export interface Attachment {
-    createdDateTime?: Date;
+export interface ApplicationProductDetailOutput {
+    avgMaterial?: MeasureOutput;
+    isCarrier?: boolean;
+    productName?: string;
+    totalMaterial?: MeasureOutput;
+}
+
+// @public (undocumented)
+export interface AttachmentListResponseOutput {
+    $skipToken?: string;
+    nextLink?: string;
+    value?: Array<AttachmentOutput>;
+}
+
+// @public (undocumented)
+export interface AttachmentOutput {
+    createdDateTime?: string;
     description?: string;
     eTag?: string;
     farmerId?: string;
     id?: string;
-    modifiedDateTime?: Date;
+    modifiedDateTime?: string;
     name?: string;
     originalFileName?: string;
     resourceId?: string;
     resourceType?: string;
     status?: string;
 }
 
-// @public (undocumented)
-export interface AttachmentListResponse {
-    nextLink?: string;
-    skipToken?: string;
-    value?: Attachment[];
-}
-
 // @public
 export interface AttachmentsCreateOrUpdate200Response extends HttpResponse {
     // (undocumented)
-    body: Attachment;
+    body: AttachmentOutput;
     // (undocumented)
     status: "200";
 }
 
 // @public
 export interface AttachmentsCreateOrUpdate201Response extends HttpResponse {
     // (undocumented)
-    body: Attachment;
+    body: AttachmentOutput;
     // (undocumented)
     status: "201";
 }
 
 // @public (undocumented)
 export interface AttachmentsCreateOrUpdateBodyParam {
     // (undocumented)
-    body?: string;
+    body?: AttachmentsCreateOrUpdateFormBody;
 }
 
 // @public
 export interface AttachmentsCreateOrUpdatedefaultResponse extends HttpResponse {
     // (undocumented)
-    body: ErrorResponse;
+    body: ErrorResponseOutput;
     // (undocumented)
     status: "500";
 }
 
 // @public (undocumented)
-export type AttachmentsCreateOrUpdateParameters = RequestParameters & AttachmentsCreateOrUpdateBodyParam;
+export interface AttachmentsCreateOrUpdateFormBody {
+    CreatedDateTime?: string;
+    Description?: string;
+    ETag?: string;
+    FarmerId?: string;
+    file?: string | Uint8Array;
```

Comment:
wonder if this should eventually support streams...

Reply:
Seems like these are form-encoded in the request, so maybe we have to collect it into a buffer anyway before sending? I'm interested in whether that can be supported without core support for doing it.

Reply:
Yeah we need to support streams. I'm planning to pick up again the RLC work to support streams now that we have the fetch client available. 

Right now I believe we'd just be able to send the file if it is loaded in memory as a Uint8Array or string, which of course is problematic for larg files.

I think I'll take on this for the March milestone